cli/internal/authentication/authentication.go

```diff
@@ -0,0 +1,116 @@
+package authentication
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+)
+
+// Reads the tokens from the storage and returns them.
+// If no tokens are present, it will return nil.
+//
+// If an error occurs while reading the tokens, it tries to clear them
+// and assumes the user is not authenticated.
+func GetCurrentAuthentication() *SessionTokens {
+	tokens, err := readTokens()
+	if err != nil {
+		// Unable to read tokens, clear them and
+		// assume that the user was not logged in.
+		clearTokens()
+		return nil
+	}
+
+	return tokens
+}
+
+// Performs a token refresh request and saves the new tokens.
+// If the refresh token has expired, the tokens will be cleared and nil will be returned.
+func RefreshTokens() (*SessionTokens, error) {
+	tokens := GetCurrentAuthentication()
+	if tokens == nil {
+		return nil, nil
+	}
+
+	// Perform the token refresh request
+	newTokens, err := DefaultKeycloakConfig.performTokenRefreshRequest(tokens.RefreshToken)
+	if err != nil {
+		return nil, fmt.Errorf("failed to refresh tokens: %w", err)
+	}
+
+	// If the new tokens are nil and we dont have an error,
+	// it means that the refresh token has expired.
+	// In this case, we should clear the tokens and return nil.
+	if newTokens == nil {
+		clearTokens()
+		return nil, nil
```

Comment:
So this does not count as an error?

Reply:
See answer https://github.com/incompetent-hosting-provider/monorepo/pull/37#discussion_r1493795794